### ISSUE TITLE
Frontail not starting on fresh install

### ISSUE THREAD

**Issue by mstormi** [openhab/openhabian#1711]
Frontail isn't starting. I have no idea why. See https://community.openhab.org/t/oh-3-3-0-frontail-not-starting-after-fresh-installation/138200/12

This nodejs stuff has been an annoying mess ever. I inherited it but for sure would not have chosen to install Frontail.
Someone proficient with this crap please help.

I figured that I need to set NODE_ENV=/usr/lib and at least
npm install -g /usr/lib/node_modules/frontail cookie cookie-parser socket.io untildify fs-tail-stream CBuffer byline command-exists connect serve-static express-session basic-auth-connect commander daemon-fix41 universal-analytics is-docker
to make the system install all the dependencies but I have no idea why it worked earlier without.

**Comment by WolfgangSn** [comment]
I just did a trial on a fresh Debian 11 VM image.
Installed openhabian. Used openhabian-config to install openhab and frontail.
frontail was not working. 
So I removed the frontail nodejs package(s) and used modified commands that are used in openhabian-config to install the modules in the /opt/frontail directory ( npm install; npm update ).
After modifying the service file entry ( e.g. references to the module directory ) it looks like frontail can be started.

**Comment by mstormi** [comment]
> and used modified commands that are 

what mods exactly ?

**Comment by WolfgangSn** [comment]
instead of: "npm install --force -g"
I changed into the /opt/frontail directory and executed "npm install"
followed by: "npm update" instead of "npm update --force -g"

**Comment by Interstellar0verdrive** [comment]
> This nodejs stuff has been an annoying mess ever. I inherited it but for sure would not have chosen to install Frontail myself Someone proficient with this please help. @Interstellar0verdrive ?

I am sorry, I have no clues, I just messed around with the css part :(

**Comment by mstormi** [comment]
as commented on the forum please validate if the latest openHABian works for you

**Comment by WolfgangSn** [comment]
> please validate if the latest openHABian works for you
Just did a test.
The cookie package is installed in location of global modules.
Starting frontail it is not found.
Other modules are not installed.

I am wondering when the -g flag was introduced to install the packages. If it always has been there could it be that it's not possible any longer to use a link to point from /usr/lib/nodes_modules/frontail to /opt/frontail ?

My installation of frontail ( based on directory timestamps ) is from 10.08.2021. 
At that time the nodes_modules directory is located under /opt/frontail.
With the current setup this directory does not exist but it is created when I modify nodejs-apps.bash as stated in my earlier posts.

**Comment by mstormi** [comment]
For me it worked on a fresh install.
But it's totally intransparent how nodejs handle this, plus nodejs version changed.
Try setting NODE_PATH env variable via systemd Environment=...    See if that makes a difference.

**Comment by WolfgangSn** [comment]
No. The module installation itself does not work here - at least not yet.
g**gle search results also did not help till now.

**Comment by WolfgangSn** [comment]
@mstormi , 
what is the output of `npm list -g` in your case ? Does it show something like:
```
corepack
frontail -> ../../../opt/frontail
npm
```

In case I run `npm install`, `npm fix --force` followed by  `npm install -g` then the provided service file can be used.
The modules are installed in /opt/frontail/node_modules with the first command. The second takes care to update the modules with regard to the used npm version and the global install just creates a link (  /lib/node_modules/frontail  ) that points to /opt/frontail/node_modules.

**Comment by mstormi** [comment]
I believe you mean "audit fix" ? 
I've added an audit fix command to openHABian, seems to work for me.
Please check if that makes a difference for you on a fresh install.
Note you need to test with the main branch.

**Comment by WolfgangSn** [comment]
> Please check if that makes a difference for you on a fresh install.

Tested on Debian Bullseye x64 in a VM:

Frontail is running with this version of openhabian-config.
There is a difference in number of modules that are installed in the frontail node_modules directory:
118 in my install from last year;  12MB
381 in the test environment; 86MB

**Comment by mstormi** [comment]
Worked for me on RPi, too so I assume this was a proper fix.
